### DISCUSSION TITLE
log4j - CVE-2021-44228

### DISCUSSION THREAD

**Discussion by davixd** [gitblit/gitblit#1403] [Q&A] [answered]
Hello guys,
is gitblit affected by CVE-2021-44228? If yes, how to fix it?

Many thanks in advance!

Best regards
David

**Comment by flaix** [comment, marked as answer]
Hi!

No, Gitblit is not affected. It uses log4j, but version 1.2.17, [which is not affected](https://www.openwall.com/lists/oss-security/2021/12/13/2). The vulnerable log4j versions start at log4j version 2.0.

To reach a similar effect with log4j 1.2.x a distinctive custom configuration of log4j needs to be set up (via JMSAppender). Gitblit does not make use of that in it's configuration and is thus not affected.

If using Gitblit with its default settings, you will have no problem. Only if you have changed the log4j configuration in an exploitable way for your use case will you have to fix that.
Also, you [should](http://slf4j.org/log4shell.html) protect your installation against attempts to deposit an exploitable log4j configuration file.

**Reply by davixd** [reply]
Thank you very much! Here is also the confirmation about log4j 1.x: http://slf4j.org/log4shell.html.
